kotlinx-coroutines-debug/src/junit/junit5/CoroutinesTimeoutExtension.kt

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.debug.junit5
+
+import kotlinx.coroutines.debug.*
+import kotlinx.coroutines.debug.runWithTimeoutDumpingCoroutines
+import org.junit.jupiter.api.extension.*
+import org.junit.platform.commons.support.AnnotationSupport
+import java.lang.reflect.*
+import java.util.*
+import java.util.concurrent.atomic.*
+
+public class CoroutinesTimeoutException(public val timeoutMs: Long): Exception("test timed out ofter $timeoutMs ms")
```

Comment:
This exception definitely shouldn't be public.

Moreover, it seems that throwing `java.util.concurrent.TimeoutException` is enough

Reply:
I don't like `java.util.concurrent.TimeoutException` here, as its documentation says the following on the first line:
> Exception thrown when a blocking operation times out.

I don't feel that this describes our case. So, I think that a separate class is warranted after all. I agree that it shouldn't be public though, as there aren't many opportunities to catch it.